layers/ip6_test.go

```diff
@@ -428,3 +428,115 @@ func TestIPv6JumbogramDecode(t *testing.T) {
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0x3b, 0x06, 0x04, 0x02, 0x03, 0x00, 0x00, 0x00,
+	0x20, 0x01, 0x0d, 0xb8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02,
+	0x20, 0x01, 0x0d, 0xb8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x03,
+	0x20, 0x01, 0x0d, 0xb8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x04,
+}
+
+func TestPacketIPv6RoutingSerialize(t *testing.T) {
+	var serialize = make([]gopacket.SerializableLayer, 0, 2)
+	var err error
+
+	ip6 := &IPv6{}
+	ip6.Version = 6
+	ip6.NextHeader = IPProtocolIPv6Routing
+	ip6.HopLimit = 64
+	ip6.SrcIP = net.ParseIP("2001:db8::1")
+	ip6.DstIP = net.ParseIP("2001:db8::2")
+	serialize = append(serialize, ip6)
+
+	srh := &IPv6Routing{
+		RoutingType:  4,
+		SegmentsLeft: 2,
+	}
+	srh.NextHeader = IPProtocolNoNextHeader
+	srh.SourceRoutingIPs = append(srh.SourceRoutingIPs, net.ParseIP("2001:db8::2"))
+	srh.SourceRoutingIPs = append(srh.SourceRoutingIPs, net.ParseIP("2001:db8::3"))
+	srh.SourceRoutingIPs = append(srh.SourceRoutingIPs, net.ParseIP("2001:db8::4"))
+	ip6.Routing = srh
+
+	buf := gopacket.NewSerializeBuffer()
+	opts := gopacket.SerializeOptions{FixLengths: true}
+	err = gopacket.SerializeLayers(buf, opts, serialize...)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	got := buf.Bytes()
+	want := testPacketIPv6Routing
+	if !bytes.Equal(got, want) {
+		t.Errorf("IPv6Routing serialize failed:\ngot:\n%#v\n\nwant:\n%#v\n\n", got, want)
+	}
+}
+
+func TestPacketIPv6RoutingDecode(t *testing.T) {
+	ip6 := &IPv6{
+		BaseLayer: BaseLayer{
+			Contents: []byte{
+				0x60, 0x00, 0x00, 0x00, 0x00, 0x38, 0x2b, 0x40, 0x20, 0x01, 0x0d, 0xb8, 0x00, 0x00, 0x00, 0x00,
+				0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x20, 0x01, 0x0d, 0xb8, 0x00, 0x00, 0x00, 0x00,
+				0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02,
+			},
+			Payload: []byte{},
+		},
+		Version:      6,
+		TrafficClass: 0,
+		FlowLabel:    0,
+		Length:       56,
+		NextHeader:   IPProtocolIPv6Routing,
+		HopLimit:     64,
+		SrcIP:        net.IP{0x20, 0x01, 0x0d, 0xb8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01},
+		DstIP:        net.IP{0x20, 0x01, 0x0d, 0xb8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02},
+	}
+
+	rt := &ip6.rt
+	rt.ipv6ExtensionBase = ipv6ExtensionBase{
+		BaseLayer: BaseLayer{
+			Contents: []byte{
+				0x3b, 0x06, 0x04, 0x02, 0x03, 0x00, 0x00, 0x00,
+				0x20, 0x01, 0x0d, 0xb8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02,
+				0x20, 0x01, 0x0d, 0xb8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x03,
+				0x20, 0x01, 0x0d, 0xb8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x04,
+			},
+			Payload: []byte{},
+		},
+		NextHeader:   IPProtocolNoNextHeader,
+		HeaderLength: uint8(6),
+		ActualLength: 56,
+	}
+	rt.RoutingType = 4
+	rt.SegmentsLeft = 2
+	rt.LastEntry = 3
```

Comment:
```suggestion
	rt.LastEntry = 2
```
idem: only 3 addresses => last index is 2